scripts/fileDigest.js

```diff
@@ -0,0 +1,265 @@
+ * @returns A function, that takes a callback passing in the filename used. This function must return a promise.
+ *
+ * @example
+ * const execute = executeWithTempFile(someContent);
+ * await execute((file) => doSomethingWithFilePath(file));
+ */
+const executeWithTempFile = (
+    content,
+    filename = Date.now().toString()
+) => async (cb) => {
+    writeTempFile(content, filename);
+    await cb(getTempFile(filename));
+    removeTempFile(filename);
+};
+
+/**
+ * Verify signature with remote (published) public key
+ *
+ * @param {*} file path to file to verify
+ * @param {*} sigFile path to file containing signature
+ */
+async function verifyRemote(file, sigFile) {
+    console.log('\nVerification of signature with remote public key:');
+
+    try {
+        const content = await getPublicKey();
+        await executeWithTempFile(content)((key) =>
+            verifySignature(key, file, sigFile)
+        );
+    } catch (e) {
+        console.error(e);
+    }
+}
+
+/**
+ * Produce signature of file checksum.
+ *
+ * The signature can be verified by running the following command:
+ * openssl dgst -<hash-algorithm> -verify <pubkey-file> -signature <signature-file> <file>
+ * (e.g. openssl dgst -sha256 -verify ./Downloads/concordium-desktop-wallet.pem -signature ./Downloads/concordium-desktop-wallet-1.0.0.dmg.hash.sig ./Downloads/concordium-desktop-wallet-1.0.0.dmg)
+ */
+async function writeSignature(file, shouldVerify = false) {
+    const sigOutFile = `${file}.sig`;
+
+    await promisify(exec)(
+        `openssl dgst -${hashAlgorithm} -sign ${publicKeyPath} -out ${sigOutFile} ${file}`
+    );
+
+    console.log('Wrote sig successfully to file:', sigOutFile);
+
+    if (!shouldVerify) {
+        return;
+    }
+
+    if (verifyKeyPath) {
+        verifySignature(verifyKeyPath, file, sigOutFile);
+    } else if (!skiprv) {
+        verifyRemote(file, sigOutFile);
+    }
+}
+
+const linuxTargets = build.linux.target;
+const winTargets = ['exe'];
+const macTargets = ['dmg', 'zip'];
+
+function getTargetsFromArgs() {
+    return [
+        [linux, linuxTargets],
+        [mac, macTargets],
+        [win, winTargets],
+    ]
+        .filter(([include]) => include)
+        .map(([, targets]) => targets)
+        .flat(2);
+}
+
+let extensions = [...linuxTargets, ...winTargets, ...macTargets];
+
+if (mac || win || linux) {
+    extensions = getTargetsFromArgs();
+}
+
+const filesToDigest = inputFile
+    ? [inputFile]
+    : extensions
+          .map((e) => path.resolve(dir, `${app.name}-${appVersion}.${e}`))
+          .filter(fs.existsSync);
+
+(async () => {
+    if (!filesToDigest.length) {
+        console.error(
+            'Found no files for specified runtime arguments in directory:',
+            dir
+        );
+    }
+    for (let i = 0; i < filesToDigest.length; i += 1) {
+        const file = filesToDigest[i];
+        const shouldVerify = i === filesToDigest.length - 1;
```

Comment:
This means that we only verify the final file?

Reply:
Yes, the assumption is, that because everything is signed in one batch with the same key, it's redundant to verify every file as the verification is only used to prevent errors with uploading incorrectly signed binaries.

Reply:
I don't see any reason for this optimization, so I believe the check should be made for every file.

Reply:
Alright, I removed it.